LASTA-OUTDOOR/lasta
# Testing/preferences view model

## Description
This pull request enhances the test coverage for the `PreferencesViewModel` class by adding comprehensive tests for each method, ensuring that all preferences are correctly updated and cleared.

## Changes Made
- Added individual tests for each preference field in the `updateUserInfo` method to ensure accurate updating of user info preferences.
- Ensured complete coverage for the `updateHikingLevel` method.
- Verified the functionality of the `clearPreferences` method by testing its ability to clear all preferences.

## Test Coverage
- Verified the correct updating of the `isLoggedIn` preference in the `updateIsLoggedIn` method.
- Tested the updating of user info preferences (`userId`, `userName`, `email`, `profilePictureUrl`, `hikingLevel`) in the `updateUserInfo` method.
- Ensured proper updating of the `hikingLevel` preference in the `updateHikingLevel` method.
- Validated the clearing of all preferences in the `clearPreferences` method.

## How to Test
1. Run the provided unit tests to ensure all methods of the `PreferencesViewModel` class are functioning as expected.
2. Verify that each preference field is correctly updated when invoking the corresponding method.
3. Test the `clearPreferences` method to confirm that all preferences are cleared.


## JaCoCo 100% coverage
![image](https://github.com/LASTA-OUTDOOR/lasta/assets/91149580/14a9eac3-6b5f-45b5-937d-984d294849d8)
